contracts/gateway/AxelarAmplifierGateway.sol

```diff
@@ -92,54 +88,65 @@ contract AxelarAmplifierGateway is IAxelarAmplifierGateway {
     |* External Functions *|
     \**********************/
 
-    function execute(SignedCommandBatch calldata signedBatch) external {
-        CommandBatch calldata batch = signedBatch.batch;
-        bytes memory data = abi.encode(batch);
-
-        // This is the EVM convention for signing non-tx data to domain separate from regular EVM txs.
-        // This should be customized per chain.
-        bytes32 batchHash = ECDSA.toEthSignedMessageHash(keccak256(data));
-
-        bool isLatestSigners = authModule.validateProof(batchHash, signedBatch.proof);
+    function verifyMessages(SignedMessageBatch calldata signedBatch) public {
+        bytes memory data = abi.encode(signedBatch.messages);
 
-        if (domainSeparator != signedBatch.batch.domainSeparator) revert InvalidDomainSeparator();
+        _verifyProof(CommandType.VerifyMessages, data, signedBatch.proof);
 
-        Command[] calldata commands = batch.commands;
+        Message[] calldata messages = signedBatch.messages;
 
-        for (uint256 i; i < commands.length; ++i) {
-            Command calldata command = commands[i];
-            bytes32 commandId = keccak256(bytes(command.messageId));
+        for (uint256 i; i < messages.length; ++i) {
+            Message calldata message = messages[i];
+            bytes32 commandId = keccak256(bytes(message.messageId));
 
             // Ignore if commandId is already executed
             if (isCommandExecuted(commandId)) {
                 continue;
             }
 
-            if (command.commandType == CommandType.ApproveContractCall) {
-                _approveContractCall(commandId, command);
-            } else if (command.commandType == CommandType.RotateSigners) {
-                if (!isLatestSigners) {
-                    continue;
-                }
-
-                isLatestSigners = false;
-
-                _rotateSigners(command.params);
-            } else {
-                revert InvalidCommand();
-            }
+            _approveMessage(commandId, message);
 
             _storage().commands[commandId] = true;
```

Comment:
Does it Make sense to cache storage slot into a variable? Especially in a loop

Reply:
it was more expensive when i cached it. I think the compiler optimizes this better due to inlining. I'll do an optimization pass again at the end